### ISSUE TITLE
SSR: Second chunk missing some styles

### ISSUE THREAD

**Issue by ka2jun8** [styled-components/styled-components#1982]
## Environment

``` package.json
"name": "styled-components",
"version": "3.4.5"
```

## Reproduction
add to ssr.test.js

``` src/test/ssr.test.js
  it('should interleave styles with rendered HTML when chunked streaming', () => {
    injectGlobal`
      body { background: papayawhip; }
    `
    const Heading = styled.h1`
      color: red;
    `

    const Body = styled.div`
      color: blue;
    `

    const SideBar = styled.div`
      color: yellow;
    `

    const Footer = styled.div`
      color: green;
    `

    const sheet = new ServerStyleSheet()
    const jsx = sheet.collectStyles(
      <div>
        <Heading>Hello SSR!</Heading>
        <Body>
        {new Array(1000)
          .fill(0)
          .map(() => <div>*************************</div>)}
        </Body>
        <SideBar>SideBar</SideBar>
        <Footer>Footer</Footer>
      </div>
    )
    const stream = sheet.interleaveWithNodeStream(renderToNodeStream(jsx))

    return new Promise((resolve, reject) => {
      let received = ''

      stream.on('data', chunk => {
        received += chunk
      })

      stream.on('end', () => {
        expect(received).toMatchSnapshot()
        expect(sheet.closed).toBe(true)
        expect(received).toMatch(/yellow/)
        expect(received).toMatch(/green/) // <- failed
        resolve()
      })

      stream.on('error', reject)
    })
  })
```

## Steps to reproduce

`jest -u src/test/ssr.test.js`

## Expected Behavior

The test should pass.

## Actual Behavior

The test is failed.

## Workaround

The workaround is here.

```
    const sheet = new ServerStyleSheet()
+   sheet.instance.tagMap = {}
    const jsx = sheet.collectStyles(
```

**Comment by mxstbr** [comment]
Mind submitting that failing tests as a PR? That'd be very helpful, thank you for the reproduction!

**Comment by probablyup** [comment]
Against v4 (develop) preferably

**Comment by probablyup** [comment]
@ka2jun8 should be fixed in v3.4.6
